packages/app/src/cli/models/extensions/specifications/flow_action.ts

```diff
@@ -1,35 +1,67 @@
 import {BaseSchema} from '../schemas.js'
 import {createExtensionSpecification} from '../specification.js'
+import {
+  validateNonCommerceObjectShape,
+  startsWithHttps,
+  validateCustomConfigurationPageConfig,
+} from '../../../services/Flow/validation.js'
+import {serializeFields} from '../../../services/Flow/serializeFields.js'
 import {joinPath} from '@shopify/cli-kit/node/path'
 import {glob, readFile} from '@shopify/cli-kit/node/fs'
-
 import {zod} from '@shopify/cli-kit/node/schema'
 
 const FlowActionExtensionSchema = BaseSchema.extend({
   name: zod.string(),
+  description: zod.string().optional(),
   type: zod.literal('flow_action'),
-  task: zod.object({
-    title: zod.string(),
-    description: zod.string(),
-    url: zod.string(),
-    validation_url: zod.string().optional(),
-    custom_configuration_page_url: zod.string().optional(),
-    custom_configuration_page_preview_url: zod.string().optional(),
-    schema: zod.string().optional(),
-    return_type_ref: zod.string().optional(),
-    fields: zod
-      .array(
-        zod.object({
-          id: zod.string(),
-          name: zod.string(),
-          label: zod.string(),
-          description: zod.string().optional(),
-          required: zod.boolean(),
-          ui_type: zod.string(),
-        }),
-      )
-      .optional(),
-  }),
+  extensions: zod
+    .array(
+      zod.object({
+        runtime_url: zod.string().url().refine(startsWithHttps),
+        validation_url: zod.string().url().refine(startsWithHttps).optional(),
+        config_page_url: zod.string().url().refine(startsWithHttps).optional(),
+        config_page_preview_url: zod.string().url().refine(startsWithHttps).optional(),
+        schema: zod.string().optional(),
+        return_type_ref: zod.string().optional(),
+      }),
+    )
+    .min(1),
+  settings: zod
+    .object({
+      fields: zod
+        .array(
+          zod
+            .object({
+              key: zod.string().optional(),
+              name: zod.string().optional(),
+              description: zod.string().optional(),
+              required: zod.boolean().optional(),
+              type: zod.string(),
+            })
+            .refine((field) => validateNonCommerceObjectShape(field, 'flow_action')),
+        )
+        .optional(),
+    })
+    .optional(),
+}).refine((config) => {
+  const {extensions} = config
+  const extension = extensions[0]
+
+  if (!extension) {
+    throw new zod.ZodError([
+      {
+        code: zod.ZodIssueCode.custom,
+        path: ['extensions'],
+        message: 'A flow action must have at least one extension',
```

Comment:
Nop, nothing is localized in the CLI

Reply:
Chatted about this offline and this is used for typescript to know that `extension` does exist in this part of the code (i.e. this is used to type `extensions[0]` to be of the `Extension` shape and no `Extension | undefined`). The error should be caught by `zod` unless there is something awfully wrong in which case it will be caught here as well.

Reply:
Decided to update this code to use `const extension = extensions[0]!`.

It's not great to coerce the typing like this but also the extension should technically be there re:zod validation. Let me know what you think.